src/main/java/org/jboss/modules/JDKModuleFinder.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.modules;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.jboss.modules.filter.PathFilters;
+
+/**
+ * A module finder which finds a standard JDK module, a module on the JDK module path, or the special module
+ * {@code org.jboss.modules}.
+ */
+public final class JDKModuleFinder implements IterableModuleFinder {
+
+    private final ConcurrentHashMap<String, FutureSpec> modules = new ConcurrentHashMap<>();
+    private final List<DependencySpec> javaSeDeps;
+    private final List<String> moduleNames;
+
+    private static final JDKModuleFinder INSTANCE = new JDKModuleFinder();
+
+    private JDKModuleFinder() {
+        List<DependencySpec> deps = new ArrayList<>();
+        for (String dep : Arrays.asList(
+            "java.compiler",
+            "java.datatransfer",
+            "java.desktop",
+            "java.instrument",
+            "java.logging",
+            "java.management",
+            "java.management.rmi",
+            "java.naming",
+            "java.prefs",
+            "java.rmi",
+            "java.scripting",
+            "java.security.jgss",
+            "java.security.sasl",
+            "java.sql",
+            "java.sql.rowset",
```

Comment:
I see you ommited java.transaction.xa - is it intentional?

Reply:
Yes.  The set is meant to mimic the set of Java 9 modules, minus the ``java.se.ee`` ones.  The ``java.transaction.xa`` module does not exist in Java 9 or 10.